hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/utils/shims/HiveCompatibleShimAbove23.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.hadoop.utils.shims;
+
+import java.io.IOException;
+import java.lang.reflect.Method;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapred.JobConf;
+
+/**
+ * When the hive version is greater than or equal to hive version 2.3
+ */
+public class HiveCompatibleShimAbove23 implements HiveCompatibleShim {
+
+  private static HiveCompatibleShimAbove23 INSTANCE = new HiveCompatibleShimAbove23();
+
+  public static HiveCompatibleShim getInstance() {
+    return INSTANCE;
+  }
+
+  @Override
+  public void invokePushProjectAndFilters(JobConf job, Class<?> inputFormatClass, Path splitPath)
+      throws IOException {
+    try {
+      Class hiveInputFormatClass = this.getClass().getSuperclass();
+      Method pushProjectionsAndFilters = hiveInputFormatClass.getDeclaredMethod("pushProjectionsAndFilters", JobConf.class,

```

Comment:
The default version of hive is 2.3, can we extends `HiveInputFormat` and call `pushProjectionsAndFilters ` directly.

Reply:
Oh, sorry, I know your concern, can this be compiled with -Dhive.version=2.2.0

Reply:
I'll test this place